### PR TITLE
Vagrant updates

### PR DESCRIPTION
**Pull request by jtslear**
Please note that some of this stuff, I ponder what impact it is going to have on systems that aren't OSX and running virtualbox.  

Also I'm pretty sure some of the ruby stuff I've done can be combined.

And lastly I opened vagrant-landrush/landrush#174, I was having issues pushing to the docker registry, hence the hack into iptables.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -78,10 +78,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     if use_atomic
       if use_fedora
         box_type = 'fedora/23-atomic-host'
-        non_atomic_box = 'fedora/23-cloud-base'
       else
         box_type = 'centos/atomic-host'
-        non_atomic_box = 'centos/7'
```
Comment (detiber) [inline Vagrantfile:84]: This is needed for the utility host when deploying CentOS/atomic-host

**Comment by detiber** [comment]
@abutcher any chance you could take a look at this?

**Comment by abutcher** [comment]
Running now on f23. @jmashburn is also giving this a run

**Comment by abutcher** [comment]
Single master working pretty well. The registry PV came up with `master1` as the server -- we probably need to set `openshift_hosted_registry_storage_host` to ensure it gets the right value.

```
15m       5m        46        docker-registry-2-jpplm   Pod                 Warning   FailedMount   {kubelet openshift-master1.example.com}   Unable to mount volumes for pod "docker-registry-2-jpplm_default(bf884194-0721-11e6-ad44-525400b263eb)": Mount failed: exit status 32
Mounting arguments: master1:/exports/registry /var/lib/origin/openshift.local.volumes/pods/bf884194-0721-11e6-ad44-525400b263eb/volumes/kubernetes.io~nfs/registry-volume nfs []
Output: mount.nfs: Failed to resolve server master1: No address associated with hostname
```

**Pull request merged**